x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardOnBinaryDvQuery.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.wildcard.mapper;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.DocValues;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.ConstantScoreScorer;
+import org.apache.lucene.search.ConstantScoreWeight;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreMode;
+import org.apache.lucene.search.Scorer;
+import org.apache.lucene.search.TwoPhaseIterator;
+import org.apache.lucene.search.Weight;
+import org.apache.lucene.search.WildcardQuery;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.automaton.Automaton;
+import org.apache.lucene.util.automaton.ByteRunAutomaton;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Query that runs a wildcard pattern across all binary doc values. 
+ * Expensive to run so normally used in conjunction with more selective query clauses.
+ */
+public class WildcardOnBinaryDvQuery extends Query {
+
+    private final String field;
+    private final String wildcardPattern;
+
+    public WildcardOnBinaryDvQuery(String field, String wildcardPattern) {
```

Comment:
Maybe take an Automaton right away, this way we can support fuzzy/regexp queries?

Reply:
Does that mean I should also change this class name to AutomatonQueryOnBinaryDv ?

Reply:
+1